### PR TITLE
feat(sdk-core): generate ecdsa challenge with safe primes in node & browser 

### PR DESCRIPTION
**Pull request by zahin-mohammad**
## Description

This PR introduces an openssl web-assembly wrapper class. Currently the only exposed functionality is generating a safe prime number which is needed during ecdsa tss challenge generation. A web-assembly module is introduced to maintain compatibility with existing node versions as much as possible while also being browser compatible. 

## Issue Number

TICKET: BG-72158

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Unit tests + manually testing via web-demo and node scripts.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code compiles correctly for both Node and Browser environments
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My commits follow [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) and I have properly described any BREAKING CHANGES
- [x] The ticket or github issue was included in the commit message as a reference
- [x] I have made corresponding changes to the documentation and on any new/updated functions and/or methods - [jsdoc](https://jsdoc.app/)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by pranavjain97** [comment]
Generated 0be3a8faef1d1cec7ba529df693658fc8769a580a208e706edab0c520c6fe6a4 as the shasum during the wasm ceremony.

**Comment by islamaminBitGo** [comment]
Witnessed ceremony, looks good.

**Comment by zahin-mohammad** [comment]
> Generated 0be3a8faef1d1cec7ba529df693658fc8769a580a208e706edab0c520c6fe6a4 as the shasum during the wasm ceremony.

Verified generation via call with pranav for sha256sum `0be3a8faef1d1cec7ba529df693658fc8769a580a208e706edab0c520c6fe6a4`

File: modules/sdk-coin-avaxc/src/avaxc.ts [hunk 1]
```diff
@@ -748,7 +748,10 @@ export class AvaxC extends BaseCoin {
     const tx = await txBuilder.build();
     const payload = tx.signablePayload;
     const signatures = tx.signature.map((s) => Buffer.from(s, 'hex'));
-    const recoverPubky = signatures.map((s) => AvaxpLib.Utils.recoverySignature(_.get(tx, '_network'), payload, s));
+    const network = _.get(tx, '_network');
+    const recoverPubky = signatures.map((s) =>
+      AvaxpLib.Utils.recoverySignature(network as unknown as AvalancheNetwork, payload, s)
+    );
```
Comment (zahin-mohammad) [inline modules/sdk-coin-avaxc/src/avaxc.ts:754]: Weird that I had to make this change 🤔  but talked through with the avaxc team for this.

File: modules/sdk-core/src/openssl/README.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Updating openssl.ts
+
+1. Clone the following [repo](https://github.com/zahin-mohammad/openssl-wasm)
+2. Use the `build.sh` script to generate the `openssl.wasm` and `openssl.js` file
+3. Convert the `openssl.js` file to a valid `openssl.ts`
```
Comment (islamaminBitGo) [inline modules/sdk-core/src/openssl/README.md:5]: Not sure if it's worth it but might be nice to put a few directions on what this conversion would look like. What functions do we need to keep in the .js files? Which ones are we removing?

Comment (zahin-mohammad) [reply modules/sdk-core/src/openssl/README.md:5]: Good point. I can add this in a follow up PR.

Comment (zahin-mohammad) [reply modules/sdk-core/src/openssl/README.md:5]: Can probably create a patch file for this directly.

**Review by islamaminBitGo — DISMISSED** [review]
Just the modulus comment change please, other one (docs comment) could be handled in a separate PR.

**Review by pengyuc-bitgo — APPROVED** [review]

**Review by islamaminBitGo — APPROVED** [review]

**Comment by zahin-mohammad** [comment]
Please merge once this has been approved.

**Review by wesgraham — APPROVED** [review]

**Review by CapnMigraine — APPROVED** [review]

**Pull request merged**